projects/cm_mcu/CommandLineTask.c

```diff
@@ -90,6 +90,33 @@ static BaseType_t clock_ctl(int argc, char **argv, char* m)
   return pdFALSE;
 }
 
+#ifdef REV2
+// this command takes one argument (from triplet version but will take two argument to include an input from config versions for octlet eeprom)
+static BaseType_t init_load_clock_ctl(int argc, char **argv, char* m)
+{
+  int copied = 0;
+  int status = -1; // shut up clang compiler warning
+  char *clk_ids[5] = {"r0a","r0b","r1a","r1b","r1c"};
+  BaseType_t i = strtol(argv[1], NULL, 10);
+  if (i < 0 || i > 5) {
+    copied +=
+        snprintf(m + copied, SCRATCH_SIZE - copied,
+            "Invalid clock chip %ld , the clock id options are r0a:0, r0b:1, r1a:2, r1b:3 and r1c:4 \r\n", i);
+    return pdFALSE;
+  }
+  copied += snprintf(m + copied, SCRATCH_SIZE - copied, "%s is programming clock %s. \r\n", argv[0], clk_ids[i]);
+  status = init_load_clk(i); // status is 0 if all registers can be written to a clock chip. otherwise, it implies that some write registers fail in a certain list.
+  if (status == 0){
+    copied += snprintf(m + copied, SCRATCH_SIZE - copied,
+        "clock synthesizer with id %s successfully programmed. \r\n", clk_ids[i]);
+  }
+  else {
+    copied += snprintf(m + copied, SCRATCH_SIZE - copied, "%s operation failed \r\n", argv[0]);
```

Comment:
maybe print out the value of the status code, if it is meaningful?

Reply:
This should be taken care by log error in LocalTasks.c I think? because it can print out the status code. This part will be implemented from your suggestion on logging the error of write (https://github.com/apollo-lhc/cm_mcu/pull/131#discussion_r858899540)